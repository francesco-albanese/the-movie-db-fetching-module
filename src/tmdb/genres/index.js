import { tmdbClient, urls } from '#tmdb/tmdb.const'
import { transformAllGenresResponse } from '#transformation'

import { parseStringifiedData } from '#utils'

export const getGenres = async(language) => {
  
  try {
    const data  = await tmdbClient.get(urls.genres, {
      params: {
        language
      },
      transformResponse: [
        data => parseStringifiedData(data),
        ({ genres }) => transformAllGenresResponse(genres)
      ]
    })

    return data
  }

  catch (e) {
    throw e
  }
}