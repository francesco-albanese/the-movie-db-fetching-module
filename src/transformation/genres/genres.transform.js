import { isEmpty } from 'lodash-es'

export const transformAllGenresResponse = (data = []) => {

  if (isEmpty(data)) {
    return []
  } 
  
  return data.map(genre => {

    const { id, name } = genre

    return {
      id,
      name
    }

  })
}