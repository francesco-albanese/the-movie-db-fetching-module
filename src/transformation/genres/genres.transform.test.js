import { transformAllGenresResponse } from './genres.transform'
import * as Fixtures from './__fixtures__/genres.transform.fixtures'

describe('transformAllGenresResponse', () => {
  it('returns an empty array if data is missing', () => {
    expect(transformAllGenresResponse()).toEqual([])
  })

  it('returns transformed genres', () => {
    const result = transformAllGenresResponse(Fixtures.data)
    expect(result).toMatchSnapshot()
  })
})