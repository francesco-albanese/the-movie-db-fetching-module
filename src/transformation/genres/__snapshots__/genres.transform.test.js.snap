// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformAllGenresResponse returns transformed genres 1`] = `
Array [
  Object {
    "id": 28,
    "name": "Action",
  },
  Object {
    "id": 12,
    "name": "Adventure",
  },
  Object {
    "id": 16,
    "name": "Animation",
  },
  Object {
    "id": 35,
    "name": "Comedy",
  },
  Object {
    "id": 80,
    "name": "Crime",
  },
  Object {
    "id": 99,
    "name": "Documentary",
  },
  Object {
    "id": 18,
    "name": "Drama",
  },
  Object {
    "id": 10751,
    "name": "Family",
  },
  Object {
    "id": 14,
    "name": "Fantasy",
  },
  Object {
    "id": 36,
    "name": "History",
  },
  Object {
    "id": 27,
    "name": "Horror",
  },
  Object {
    "id": 10402,
    "name": "Music",
  },
  Object {
    "id": 9648,
    "name": "Mystery",
  },
  Object {
    "id": 10749,
    "name": "Romance",
  },
  Object {
    "id": 878,
    "name": "Science Fiction",
  },
  Object {
    "id": 10770,
    "name": "TV Movie",
  },
  Object {
    "id": 53,
    "name": "Thriller",
  },
  Object {
    "id": 10752,
    "name": "War",
  },
  Object {
    "id": 37,
    "name": "Western",
  },
]
`;
